src/core/LambdaWrapper.ts

```diff
@@ -86,14 +86,14 @@ export default class LambdaWrapper<TConfig extends LambdaWrapperConfig = LambdaW
       }
     };
 
-    // If Epsagon is enabled, wrap the instance in the Epsagon wrapper
-    if (process.env.EPSAGON_TOKEN && process.env.EPSAGON_SERVICE_NAME) {
-      Epsagon.init({
-        token: process.env.EPSAGON_TOKEN,
-        appName: process.env.EPSAGON_SERVICE_NAME,
-      });
+    // If Lumigo is enabled, wrap the handler in the Lumigo wrapper
```

Comment:
I think it would help future devs understanding if we explained the difference here in set-up between auto-tracing and manual tracing (where `LUMIGO_TRACER_TOKEN` env var must be explicitly set?).

Reply:
Is this good? https://github.com/comicrelief/lambda-wrapper/pull/1168/commits/7b2949945cf1f900d58814d91b620575237127fa

It does actually prompt a different question: if we're auto-tracing, what happens if we wrap the handler again? Would be funny if we got duplicate traces for every function! I'll ask them.

Reply:
Just to summarise what Lumigo said: it is safe to double-wrap a function. But rather than do this, they suggested that we can guarantee all of our functions are auto-traced by setting the `lumigo:auto-trace` AWS tag as per [their docs](https://docs.lumigo.io/docs/serverless-applications#automatic-instrumentation).

I think that's a great idea as we need to add new tags anyway (and it means we don't need to add our Lumigo token to every pipeline), however I still want Lambda Wrapper to work with manual tracing, and in the meantime I figured out a way to distinguish an auto-traced from manually traced function – the Lumigo Lambda Layer needs configuring via some unambiguous env vars ([docs](https://docs.lumigo.io/docs/lambda-layers)).

So I've made some tweaks for the best of both worlds: if auto-traced, we won't double-wrap the handler, and if not auto-traced but the token is present, we will wrap the handler. I've moved the logic behind this into methods with their own JSDoc so that the code explains itself a bit better.

Reply:
Thanks @seb-cr. I've caught up with what Lumigo have written on the shared Slack channel and your notes / commits.

One thought I had - its more difficult for us to detect if we're auto-tracing vs manual-tracing because we're using `LUMIGO_TRACER_TOKEN` for both. What if we used something different to `LUMIGO_TRACER_TOKEN` when setting up manual-tracing?

The main reason I flag this is because `isLumigoWrappingUs()` is relying on the value of a Lumigo internal env var `AWS_LAMBDA_EXEC_WRAPPER` which could easily change overnight without us knowing.

If we simply used `LUMIGO_TOKEN` or `LUMIGO_MANUAL_TOKEN` as our env var when setting up manual tracing then we wouldn't have this problem.

I can't see anything in the docs which says we must use `LUMIGO_TRACER_TOKEN` env var name when manually tracing, but if you have please flag it up.

Reply:
True. I did consider having a separate env var. I think what I didn't like about it was how easy it is to turn on or off auto-tracing from the Lumigo UI, so I'm wanting Lambda Wrapper to work as much as possible with both, without needing manual changes. Technically I guess it would still work with both the manual and auto env vars present, but that feels like a bad configuration to be able to get into.

I don't expect the value of `AWS_LAMBDA_EXEC_WRAPPER` to change unexpectedly, as this would break tracing for everyone who had manually configured their Lambdas (which the docs explain how to do, so there must be some people who do it this way!). And, if it does change, it wouldn't break anything for us; Lambda Wrapper will just think it needs to apply the tracer itself, and tracing and tagging will continue to function.

All that aside: I suppose that given Lumigo has told us double-wrapping is safe, we don't _need_ to try to detect the Lambda layer, and can always wrap if the Lumigo token is present. That might be best in terms of simplicity and maintainability rather than me trying to overengineer it. 😁